File: 2965-find-missing-and-repeated-values/2965-find-missing-and-repeated-values.java
class Solution {
    public int[] findMissingAndRepeatedValues(int[][] grid) {
        int n = grid.length;
        int m = n * n;
        int[] freq = new int[m + 1];
        for (int[] row : grid) {
            for (int num : row) {
                freq[num]++;
            }
        }
        int repeated = -1;
        int missing = -1;
        for (int i = 1; i <= m; i++) {
            if (freq[i] == 2) {
                repeated = i;
            } else if (freq[i] == 0) {
                missing = i;
            }
        }
        return new int[]{repeated, missing};
    }
}

File: 2965-find-missing-and-repeated-values/2965-find-missing-and-repeated-values.cpp
class Solution {
public:
    vector<int> findMissingAndRepeatedValues(vector<vector<int>>& grid) {
        int n = grid.size();
        int m = n * n;

        int repeated = -1, missing = -1;

        for (int i = 1; i <= m; i++) {
            int count = 0;

            for (int r = 0; r < n; r++) {
                for (int c = 0; c < n; c++) {
                    if (grid[r][c] == i)
                        count++;
                }
            }

            if (count == 2)
                repeated = i;
            else if (count == 0)
                missing = i;
        }

        return {repeated, missing};
    }
};